analyzer/codechecker_analyzer/cmd/parse.py

```diff
@@ -467,10 +467,18 @@ def parse(plist_file, metadata_dict, rh, file_report_map):
 
     LOG.debug("Parsing input file '%s'", plist_file)
 
-    if 'result_source_files' in metadata_dict and \
-            plist_file in metadata_dict['result_source_files']:
+    result_source_files = {}
+    if 'result_source_files' in metadata_dict:
+        result_source_files = metadata_dict['result_source_files']
+    else:
+        for tool in metadata_dict.get('tools', {}):
+            result_src_files = tool.get('result_source_files', {})
+            result_source_files = dict(result_source_files.items() +
+                                       result_src_files.items())
```

Comment:
Same as above.

Reply:
`result_source_files.update(result_src_files.items())` wouldn't be enough here, if we just want to merge the two dictionaries?